### PR TITLE
Add Shell support for raw Tx Injection

### PR DESCRIPTION
**Pull request by D-Triveni**
Fixes SHEL-2292

### CODE REVIEW

File: samples/wifi/shell/src/wifi_raw_tx_pkt_shell.c [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+	int rate_flags = 0, data_rate = 0, queue_num = 0;
+
+	if (argc != 7) {
+		shell_help(sh);
+		return 1;
+	}
+
+	LOG_INF("argument count: %d", argc);
+	for (int i = 1; i < argc; i += 2) {
+		if (strcmp(argv[i], "-f") == 0 || strcmp(argv[i], "--rate-flags") == 0) {
```
[the remark this answers is not in the document]
Comment (D-Triveni) [reply samples/wifi/shell/src/wifi_raw_tx_pkt_shell.c]: No specific reason for using ``strcmp``. Updated to use ``getopt`` for consistency with the approach in ``net/l2/wifi/``

**Review by krish2718 — APPROVED** [review]

**Review by bama-nordic — APPROVED** [review]

**Review by VivekUppunda — APPROVED** [review]

File: samples/wifi/shell/README.rst [hunk 2]
```diff
@@ -220,6 +227,28 @@ It adds the following commands to interact with the :ref:`lib_wifi_credentials`
...
+       | [-d, --data-rate] : Data rate value.
+       | [-q, --queue-number] : Queue number.
+       | [-h, --help] : Print out the help for the configure command.
+   * - send
+     - | Send raw TX packets
+       | parameters:
+       | [-m, --mode] : Mode of transmission (either continuous or fixed).
+       | [-n, --number-of-pkts] : Number of packets to be transmitted.
+       | [-t, --inter-frame-delay] : Delay between frames/packets.
+       | [-h, --help] : Print out the help for the send command.
```
Comment (richabp) [inline samples/wifi/shell/README.rst:250]:
```suggestion
     - | Configure the raw TX packet header with the following parameters:
       | [-f, --rate-flags]: Rate flag value.
       | [-d, --data-rate]: Data rate value.
       | [-q, --queue-number]: Queue number.
       | [-h, --help]: Print out the help for the configure command.
   * - send
     - | Send raw TX packets
       | parameters:
       | [-m, --mode]: Mode of transmission (either continuous or fixed).
       | [-n, --number-of-pkts]: Number of packets to be transmitted.
       | [-t, --inter-frame-delay]: Delay between frames or packets.
       | [-h, --help]: Print out the help for the send command.
```

**Review by richabp — CHANGES_REQUESTED** [review]
This needs a Changelog entry for the added support.

**Comment by D-Triveni** [comment]
> This needs a Changelog entry for the added support.

Added.

**Review by rado17 — APPROVED** [review]

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 3]
```diff
@@ -472,6 +472,13 @@ Wi-Fi samples
 
 * Added the :ref:`wifi_throughput_sample` sample that demonstrates how to measure the network throughput of a Nordic Wi-Fi enabled platform under the different Wi-Fi stack configuration profiles.
 
+* :ref:`wifi_shell_sample` sample:
+
+  * Updated:
+
+    * Added `raw_tx` extension to the Wi-Fi command line.
```
Comment (richabp) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:479]:
```suggestion
    * Added ``raw_tx`` extension to the Wi-Fi command line.
```

**Review by richabp — CHANGES_REQUESTED** [review]

**Review by richabp — APPROVED** [review]

**Review by krga2022 — APPROVED** [review]

**Pull request merged**